dev/gp-gcloud/cmd/compute/instance-templates-create.go

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package compute
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+
+	"encoding/json"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/compute/v1"
+)
+
+type LocalSSD struct {
+	DeviceName string
+	Interface  string
+}
+
+func newInstanceTemplatesCreateCommand() *cobra.Command {
+	var (
+		region               string
+		project              string
+		machineType          string
+		minCpuPlatform       string
+		serviceAccount       string
+		serviceAccountScopes string
+		bootDiskSize         string
+		bootDiskType         string
+		tags                 string
+		labels               string
+		image                string
+		networkInterface     string
+		metadata             string
+		metadataFromFile     string
+		localSSDs            []string
+		noRestartOnFailure   bool
+		dryRun               bool
+	)
+
+	cmd := &cobra.Command{
+		Use:     "create",
+		Short:   "create wrapper",
+		Example: "  create instance-template-1",
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) != 1 {
+				log.Fatal("no name was provided")
+			}
+
+			name := args[0]
+
+			ctx := context.Background()
+
+			c, err := google.DefaultClient(ctx, compute.CloudPlatformScope)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			computeService, err := compute.New(c)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			networkInterfaceParsed := parseLabels(networkInterface)
+			automaticRestart := !noRestartOnFailure
+			metadata := parseMetadata(metadata)
+			metadata = append(metadata, parseMetadataFromFile(metadataFromFile)...)
+			localSSDDevices := parseLocalSSDs(localSSDs)
+			labels := parseLabels(labels)
+
+			rb := &compute.InstanceTemplate{
+				Name: name,
+				Properties: &compute.InstanceProperties{
+					MachineType:    machineType,
+					MinCpuPlatform: minCpuPlatform,
+					ServiceAccounts: []*compute.ServiceAccount{
+						{
+							Email:  serviceAccount,
+							Scopes: strings.Split(serviceAccountScopes, ","),
+						},
+					},
+					Disks: []*compute.AttachedDisk{
+						{
+							AutoDelete: true,
+							Boot:       true,
+							InitializeParams: &compute.AttachedDiskInitializeParams{
+								DiskSizeGb:  parseDiskSize(bootDiskSize, 10),
+								DiskType:    bootDiskType,
+								SourceImage: image,
+								Labels:      labels,
+							},
+						},
+					},
+					Tags: &compute.Tags{
+						Items: strings.Split(tags, ","),
+					},
+					Labels: labels,
+					NetworkInterfaces: []*compute.NetworkInterface{
+						{
+							AccessConfigs: []*compute.AccessConfig{
+								{
+									NetworkTier: networkInterfaceParsed["network-tier"],
+									Type:        "ONE_TO_ONE_NAT",
```

Comment:
How do we know this is the right type?

Reply:
that was really the only reasonable option and is default value afair

Reply:
Looks like it is the only [option](https://cloud.google.com/compute/docs/reference/rest/v1/instances/updateNetworkInterface). Thanks!